perfkitbenchmarker/pkb.py

```diff
@@ -417,6 +422,102 @@ def _PrintHelp(matches=None):
+
+* windows packages
+`./pkb.py --helpmatchmd windows_packages.*`
+
+* GCP provider:
+`./pkb.py --helpmatchmd providers.gcp.* >> testsuite_docs/providers_gcp.md`
+
+  Args:
+    matches: regex string or None. Filters help to only those whose name
+      matched the regex. If None then all flags are printed.
+  """
+
+  """ example docstring:
+  absl.app:
+  --[no]only_check_args: Set to true to validate args and exit.
+    (default: 'false')
+  --[no]pdb_post_mortem: Set to true to handle uncaught exceptions with PDB post
+    mortem.
+    (default: 'false')
+  --profile_file: Dump profile information to a file (for python -m pstats).
+    Implies --run_with_profiling.
+  --[no]run_with_pdb: Set to true for PDB debug mode
+    (default: 'false')
+  --[no]run_with_profiling: Set to true for profiling the script. Execution will
+    be slower, and the output format might change over time.
+    (default: 'false')
+  --[no]use_cprofile_for_profiling: Use cProfile instead of the profile module
+    for profiling. This has no effect unless --run_with_profiling is set.
+    (default: 'true')
+"""
+
+  # normal helpmatch search from above
+  if not matches:
+    print(FLAGS)
+  else:
+    flags_by_module = FLAGS.flags_by_module_dict()
+    modules = sorted(flags_by_module)
+    regex = re.compile(matches)
+    for module_name in modules:
+      if regex.search(module_name):
+        # where are we relative to pkb home?
+        BASE_RELATIVE = '../'
+        # matches module name
+        module_regex = r'^\s+?(.*?):.*'
+        # matches each flag
+        flags_regex = r'(^\s\s--.*?(?=^\s\s--|\Z))+?'
+        # matches flag name in each flag
+        flagsmd_regex = re.compile(r'^\s+?(--.*?)(:.*\Z)',
+                                   re.MULTILINE | re.DOTALL)
+        # matches triple quoted comments
+        docstring_regex = r'"""(.*?|$)"""'
+        # output standard help to string for markdown processing
+        txt = FLAGS.module_help(module_name)
+        # converts module name to github linkable string
+        # eg: perfkitbenchmarker.linux_benchmarks.iperf_vpn_benchmark ->
+        # perfkitbenchmarker/linux_benchmarks/iperf_vpn_benchmark.py
+        module = re.search(module_regex, txt, ).group(1)
+        module_link = module.replace('.', '/') + '.py'
+        # puts flag name in a markdown code block for visibility
+        flags = re.findall(flags_regex, txt, re.MULTILINE | re.DOTALL)
+        flags[:] = [flagsmd_regex.sub(r"`\1`\2", flag) for flag in flags]
+       # get the docstring for the module without importing everything into our
+       # namespace. Probably a better way to do this
+        docstring = 'No description available'
+        # only pull doststrings from inside pkb source files
+        if (isfile(module_link)):
+          with open(module_link, "r") as f:
+            source = f.read()
+            # get the triple quoted matches
+            dsm = re.search(docstring_regex, source, re.MULTILINE | re.DOTALL)
```

Comment:
Minor nit: dsm comes out of the blue as an abbrevation for some docstring. Perhaps call this docstring_match or just match.

Reply:
Fixed.